packages/devp2p/src/rlpx/rlpx.ts

```diff
@@ -259,17 +267,27 @@ export class RLPx extends EventEmitter {
   _refillConnections() {
     if (!this._isAlive()) return
     debug(
-      `refill connections.. queue size: ${this._peersQueue.length}, peers: ${
+      `refill connections.. (selector ${this._refillIntervalSelectionCounter}) peers: ${
         this._peers.size
-      }, open slots: ${this._getOpenSlots()}`
+      }, queue size: ${this._peersQueue.length}, open slots: ${this._getOpenSlots()}`
     )
+    // Rotating selection counter going in loop from 0..9
+    this._refillIntervalSelectionCounter = (this._refillIntervalSelectionCounter + 1) % 10
 
-    this._peersQueue = this._peersQueue.filter((item: any) => {
+    this._peersQueue = this._peersQueue.filter((item) => {
       if (this._getOpenSlots() === 0) return true
       if (item.ts > Date.now()) return true
 
-      this._connectToPeer(item.peer)
-      return false
+      // Randomly distributed selector based on peer ID
+      // to decide on subdivided execution
+      const selector = buffer2int((item.peer.id! as Buffer).slice(0, 1)) % 10
+      if (selector === this._refillIntervalSelectionCounter) {
+        this._connectToPeer(item.peer)
+        return false
+      } else {
+        // Still keep peer in queue
+        return true
+      }
```

Comment:
Out of curiosity, do you know why this technique improves connection reliability? 

With this change the connection attempts are spaced out and for any given peer there's a 1 in 10 chance it will be made?

Reply:
This distributes the networking traffic over time, so there's not all connections hammered out at once - then followed by a longer non-networking-activity - and subsequently many of the replies from the peers also coming in at a similar PIT (in the old version). Instead the traffic gets out and in more frequently and distributed. I am not 100% sure about the reliability impacts in a narrower sense, so e.g. if this directly prevents connection losses by avoiding concurrent network handling, my deeper networking experience is too limited for that. But this definitely leads to a healthier protocol flow - this long waiting period is avoided - becoming e.g. very apparent when the last peer connection gets lost and there is another 30 second wait - but there is always some activity/connection reattempts going on. I would also think that these distributes better on available protocol respectively implementation-given resource restrictions like the maximum number of allowed peer connections or the number of slots for the queue, but not completely sure here.

However benefit reasons are distributed: this is one of the changes which can be really "felt" when running the client, so the first connection pickup after start is just directly and apparently getting so much faster that it gets a bit out of question if there "might be" some positive effect. 😄